src/Forms/FormHandler.php

```diff
@@ -1495,37 +1495,46 @@ public function perform_add(string &$form_action): void
         foreach ($this->FG_EDIT_FORM_ELEMENTS as &$row) {
             $field = $row["name"] ?? "";
             $attr = $row["attributes"] ?? [];
-            if (empty($field) || array_key_exists("disabled", $attr)) {
+            if (empty($field) || array_key_exists("disabled", $attr) || !array_key_exists($field, $processed)) {
                 continue;
             }
 
-            if (array_key_exists("multiple", $attr) && is_array($processed[$field])) {
-                $values[$field] = (int)array_sum($processed[$field]);
+            $value = $processed[$field];
+            $check_empty = $row["check_empty"] ?? "";
+
+            if (array_key_exists("multiple", $attr) && is_array($value)) {
+                $values[$field] = (int)array_sum($value);
             }
+
             if (!empty($row["validator"])) {
-                if ($processed[$field] === "" && str_starts_with($row["check_empty"] ?? "", "NO")) {
+                $result = call_user_func($row["validator"], $value);
+                $row["validation_err"] = $result;
+                if ($result !== true) {
+                    $this->all_fields_valid = false;
+                    $form_action = "ask-add";
+                    continue;
+                }
+            } elseif ($value === "") {
+                if ($check_empty === "NO-NULL") {
+                    $value = null;
+                } elseif ($check_empty === "NO") {
```

Comment:
This check for $check_empty === "NO" has to happen *before* the check for the validator. If that element starts with "NO" *and* the field is empty, then we shouldn't run the validator check at all.